src/clj_jgit/porcelain.clj

```diff
@@ -461,6 +461,12 @@
      (-> repo .reset
          (.setRef ref)
          (.setMode ^ResetCommand$ResetType (reset-modes mode-sym))
+         (.call)))
+  ([^Git repo ref mode-sym path]
+     (-> repo .reset
+         (.setRef ref)
+         (.setMode ^ResetCommand$ResetType (reset-modes mode-sym))
+         (.addPath path)
```

Comment:
Would be nice to also support multiple paths here. After a cursory look at the JGit api all git commands that take single paths use the same method signature, so it would make sense to introduce a helper function `(add-cmd-paths [cmd-instance paths])` to `internal.clj` that calls `.addPath` for each `path` on the given `cmd-instance`.